cliccl: add `load show` subcommand to display backup info

madelynnblue opened the pull request:
Makes output like:

```
$ ./cockroach load show ~/load-tbl/orders.sst
StartTime: 1969-12-31 19:00:00 -0500 EST (0.000000000,0)
EndTime: 2017-09-11 15:18:22.411692227 -0400 EDT (1505157502.411692227,0)
DataSize: 281537130
Rows: 0
IndexEntries: 0
SystemRecords: 0
FormatVersion: 0
ClusterID: 00000000-0000-0000-0000-000000000000
NodeID: 0
BuildInfo: CockroachDB   (, built , )
Spans:
        /Table/5{1-2}
Files:
        1.sst:
                Span: /Table/51/1/1{/0-335875}
                Sha512: 5237eab85bf185c00652a1591cba57f5bec8e44d231c1d2c266f104141659d663e22bdaddf945bc8c4f034a800f0a8bfa08a3f2a7fd2640721e5d55502c84a10
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        2.sst:
                Span: /Table/51/1/{1335875/0-2669511}
                Sha512: 614ab69357789438d0b8f53dd4830fb9d6526a35580ae0d0033c943c88cddb9d222364150a72353da9da36e020f661e522c28a8ae0e05bf302c5ee174d
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        3.sst:
                Span: /Table/51/1/{2669511/0-4003588}
                Sha512: a852d30f23d9ffc2185b5f0c188666460030fb27e08d7f6f340878c33dca76fb45a985926945fa4689f4b1c9285c8622857811a6a129f724c798b402d3
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        4.sst:
                Span: /Table/51/1/{4003588/0-5337991}
                Sha512: e7e244ec596880334c01b82063c4b10f731a521078fa8ad9da13d8181bd92bb3f4cabdba6a49e683a37685eea423686266dce95090bfed6a64dff7f399
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        5.sst:
                Span: /Table/51/{1/5337991/0-3/8367/3808196}
                Sha512: 198cbaadc460f81c555adeaf7357e75c1608a9146cb032a8c872423e35c87dacb2125f973fd8ca26b0848bbda0c381b0d612d38f57a44a5979dc101273
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        6.sst:
                Span: /Table/51/3/{8367/3849255/0-10440/5999872/0/NULL}
                Sha512: e79d9d4a390a7dab71d4fd706f818dc014b51e8e99347df3cf588091fb7c5b502b060ee3e807d98c53e78d4964927870b57ed6240865e6957104bc3bda
                DataSize: 0
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
Descriptors:
        51: orders
```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18434)
<!-- Reviewable:end -->

danhhz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/ccl/cliccl/load.go, line 176 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4qKuNHTAOsMiZuaT6:-Ku4qKuNHTAOsMiZuaT7:b-gaesa2) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L176)):*
> ```Go
> 	start := time.Unix(0, desc.StartTime.WallTime)
> 	end := time.Unix(0, desc.EndTime.WallTime)
> 	fmt.Printf("StartTime: %s (%s)\n", start, desc.StartTime)
> ```

in your example output above, it's printing these timestamps in local time. we should be printing them in UTC which is used basically everywhere else, otherwise it'll be pretty confusing.

maybe `desc.StartTime.Format(time. RFC3339Nano)`? here and elsewhere

---

*[pkg/ccl/cliccl/load.go, line 178 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4r8aKlQzE36-eZtiF:-Ku4r8aKlQzE36-eZtiG:bmj9ggv) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L178)):*
> ```Go
> 	fmt.Printf("StartTime: %s (%s)\n", start, desc.StartTime)
> 	fmt.Printf("EndTime: %s (%s)\n", end, desc.EndTime)
> 	fmt.Printf("DataSize: %d\n", desc.EntryCounts.DataSize)
> ```

it'd be nice to humanize this too

`fmt.Printf("DataSize: %d (%s)\n", desc.EntryCounts.DataSize, humanizeutil.IBytes(desc.EntryCounts.DataSize))`

---

*[pkg/ccl/cliccl/load.go, line 195 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4rX7vUCyevcQIYFu_:-Ku4rX7vUCyevcQIYFua:bogfvj6) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L195)):*
> ```Go
> 		fmt.Printf("		Span: %s\n", f.Span)
> 		fmt.Printf("		Sha512: %0128x\n", f.Sha512)
> 		fmt.Printf("		DataSize: %d\n", f.EntryCounts.DataSize)
> ```

ditto humanize

---

*[pkg/ccl/cliccl/load.go, line 202 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4sAnrRQvEy4iI1EMT:-Ku4sAnrRQvEy4iI1EMU:bc5vjjy) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L202)):*
> ```Go
> 	fmt.Printf("Descriptors:\n")
> 	for _, d := range desc.Descriptors {
> 		if desc := d.GetTable(); desc != nil {
> ```

might also be nice to print the databases, too

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-:-Ku4sLIAjDIBItn2yCs1:bnfp4nl)*
<!-- Sent from Reviewable.io -->

dianasaur323 commented:
@mjibson did you want to make this a user-facing command? or is this just for debugging purposes?

danhhz commented:
Just for debugging

madelynnblue committed:
cliccl: add `load show` subcommand to display backup info

Example output:

```
StartTime: 1970-01-01T00:00:00Z (0.000000000,0)
EndTime: 2017-09-12T19:04:53.74638877Z (1505243093.746388770,0)
DataSize: 211103038 (201 MiB)
Rows: 0
IndexEntries: 0
SystemRecords: 0
FormatVersion: 0
ClusterID: 00000000-0000-0000-0000-000000000000
NodeID: 0
BuildInfo: CockroachDB   (, built , )
Spans:
        /Table/5{1-2}
Files:
        1.sst:
                Span: /Table/51/1/1{/0-902791/0/NULL}
                Sha512: 0ad2effe72d9fc5bd03da616f5fc586c8e272208e524dde118beb6e6d649d7760c7842afa8c2a9aae612c39ba19974f5cf178e48ff04b82692e443a68545b379
                DataSize: 0 (0 B)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        2.sst:
                Span: /Table/51/1/{1902816/0-3805765/0/NULL}
                Sha512: b570a22f494b2e3ac553d1cfd1ed559378f3177ed470d8baced7137fb3f8a95a6efa9cbce1d0981324a05dba2c5708feb30fe2d862815cb485972d2df2724372
                DataSize: 0 (0 B)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        3.sst:
                Span: /Table/51/1/{3805766/0-5709250/0/NULL}
                Sha512: 938cb2e531e164d660a8a3e9ddc2603e343061f860efcd4a29a74b5f9f6a38e7fe1ace78b13b5aee5dab2a39457c0887bc827cf7e76b76a3f1d284e32c54a332
                DataSize: 0 (0 B)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        4.sst:
                Span: /Table/51/{1/5709251/0-3/10292/3714624/0/NULL}
                Sha512: 539a9a0473e92d3ae53460b63e4a9e5c90461d4b6af861c897b1ef6469d317a91857bf6dd8762d268a49c1ac77cc564c32fb24c17603fde68c31c9f7fba817d8
                DataSize: 0 (0 B)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
        5.sst:
                Span: /Table/51/3/10{292/3722688/0-440/5999872/0/NULL}
                Sha512: 6bb4feeec8394661958222a26fdbf20ec9a2a155a8c45494eade9dc2025b64040e7f698de4a17e55619544fddc3170b401c1cf2987ddcdf9346606a2008bfd87
                DataSize: 0 (0 B)
                Rows: 0
                IndexEntries: 0
                SystemRecords: 0
Descriptors:
        50: csv (database)
        51: orders (table)
```

mjibson commented:
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/ccl/cliccl/load.go, line 176 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4qKuNHTAOsMiZuaT6:-Ku5sTbhzyUtie4flfo7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L176)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

in your example output above, it's printing these timestamps in local time. we should be printing them in UTC which is used basically everywhere else, otherwise it'll be pretty confusing.

maybe `desc.StartTime.Format(time. RFC3339Nano)`? here and elsewhere
</blockquote></details>

Done.

---

*[pkg/ccl/cliccl/load.go, line 178 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4r8aKlQzE36-eZtiF:-Ku5s_eveTVMWsAGsxly:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L178)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

it'd be nice to humanize this too

`fmt.Printf("DataSize: %d (%s)\n", desc.EntryCounts.DataSize, humanizeutil.IBytes(desc.EntryCounts.DataSize))`
</blockquote></details>

Done.

---

*[pkg/ccl/cliccl/load.go, line 195 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4rX7vUCyevcQIYFu_:-Ku5sgKVXN9zodyh3Rz1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L195)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

ditto humanize
</blockquote></details>

Done.

---

*[pkg/ccl/cliccl/load.go, line 202 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434#-Ku4sAnrRQvEy4iI1EMT:-Ku5sx0Xq0aISlKzHueV:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/03bc83167bc06560efcc443ed582f392a65974a9/pkg/ccl/cliccl/load.go#L202)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

might also be nice to print the databases, too
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18434)*
<!-- Sent from Reviewable.io -->
